### PR TITLE
feat: Added tutorial to scraper website from multiple Actor runs

### PR DESCRIPTION
**Pull request by drobnikj**
* Added tutorial about how to scrape a single website from multiple Actor runs
* Added example of code under `./examples`

I do it as short as possible on one side, plus cover all necessary problems.
Of course, it is not a production-ready solution, but it should give the reader an idea of how to create a scraper running from multiple Actor runs.

### CODE REVIEW

File: sources/academy/tutorials/node_js/multiple-runs-scrape.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
+---
+title: Scrape website from multiple Actor runs
```
Comment (fnesveda) [inline sources/academy/tutorials/node_js/multiple-runs-scrape.md:2]: Let's rename this file to `parallel-scraping.md`
```suggestion
title: Scrape website in parallel with multiple Actor runs
```

**Review by fnesveda — CHANGES_REQUESTED** [review]
Looks good, I just had some suggestions about naming and structure.

I would just rename "Master" to "orchestrator" to make it more intuitive, and change the folder structure of the examples to this:
```
- examples/
    - parallel-scraping/
        - orchestrator/
            - .actor/
            - src/
            - ...
        - scraper/
            - .actor/
            - src/
            - ...
```

**Comment by fnesveda** [comment]
@metalwarrior665 Could you or someone from the delivery team who has more experience writing Actors than us review this, if it's OK from your point of view?

**Comment by drobnikj** [comment]
> "Master" to "orchestrator"

I was afraid some renaming would happen, and I will need it to replace everywhere 😅

File: sources/academy/tutorials/node_js/multiple-runs-scrape.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+import MasterActorMainTs from '!!raw-loader!../../../../examples/ts-parallel-runs-scraping-master/src/main.ts';
+import MasterActorInputSchemaJson from '!!raw-loader!../../../../examples/ts-parallel-runs-scraping-master/.actor/input_schema.json';
+import ScraperActorMainTs from '!!raw-loader!../../../../examples/ts-parallel-runs-scraping-scraper/src/main.ts';
+import ScraperActorInputSchemaJson from '!!raw-loader!../../../../examples/ts-parallel-runs-scraping-scraper/.actor/input_schema.json';
+
+Imagine a large website that you need to scrape. You have a scraper that works well, but scraping the whole website is slow.
+You can speed up the scraping process by running multiple instances of the scraper in parallel.
+This tutorial will guide you through setting up your scraper to run multiple instances in parallel.
+
+> In a rush? You can check [full code example](https://github.com/apify/apify-docs/tree/master/examples/ts-parallel-runs-scraping-master) right away.
```
Comment (TC-MO) [inline sources/academy/tutorials/node_js/multiple-runs-scrape.md:26]:
```suggestion
:::note In a rush? 

You can check [full code example](https://github.com/apify/apify-docs/tree/master/examples/ts-parallel-runs-scraping-master) right away.

:::
```

I would suggest refraining from using codeblock this way, admonitions are much more elegant solution for this kind of notes.

File: sources/academy/tutorials/node_js/multiple-runs-scrape.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+The Master Actor orchestrates the parallel execution of scraper Actor runs.
+It runs multiple instances of the scraper Actor and passes the request queue and dataset to them.
+For the Actor's base structure, we use Apify CLI and create a new Actor with the following command and use the [Empty TypeScript Actor template](https://apify.com/templates/ts-empty).
+
+```bash
+apify create master-actor
+````
+
+> If you don't have Apify CLI installed, you can find the installation instructions [here](https://docs.apify.com/cli).
```
Comment (TC-MO) [inline sources/academy/tutorials/node_js/multiple-runs-scrape.md:43]:
```suggestion
If you don't have Apify CLI installed, you can find the installation instructions [here](https://docs.apify.com/cli).
```

File: sources/academy/tutorials/node_js/multiple-runs-scrape.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+    }
+    state.isInitialized = true;
+}
+```
+
+Once Actor is initialized, it launches parallel scraper runs and waits for them to complete using `Promise.all()`.
+Additionally, by registering for abort events, the Actor can terminate all parallel runs if the Coordinator Actor is stopped.
+
+<Tabs groupId="main">
+<TabItem value="main.ts" label="main.ts">
```
Comment (TC-MO) [inline sources/academy/tutorials/node_js/multiple-runs-scrape.md:141]: nitpick, could we make sure that codetabs are in the same order throughout the whole doc? It's both for consistency sake and due to the fact that docusaurus changes tab in whole documents if it has the same value

File: sources/academy/tutorials/node_js/multiple-runs-scrape.md [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+
+### Pushing to Apify
+
+Once you have the Master Actor ready, you can push it to Apify using the following command from the root directory of the Actor project:
+
+```bash
+apify push
+```
+
+> If you are pushing the Actor for the first time, you will need to [login to your Apify account](https://docs.apify.com/cli/docs/reference#apify-login).
```
Comment (TC-MO) [inline sources/academy/tutorials/node_js/multiple-runs-scrape.md:157]:
```suggestion
:::tip First login

If you are pushing the Actor for the first time, you will need to [login to your Apify account](https://docs.apify.com/cli/docs/reference#apify-login).

:::
```

File: sources/academy/tutorials/node_js/multiple-runs-scrape.md [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+<Tabs groupId="main">
+<TabItem value="main.ts" label="main.ts">
+<CodeBlock language="typescript">{ScraperActorMainTs}</CodeBlock>
+</TabItem>
+<TabItem value="input_schema.json" label="input_schema.json">
+<CodeBlock language="json">{ScraperActorInputSchemaJson}</CodeBlock>
+</TabItem>
+</Tabs>
+
+> You can check [full code example](https://github.com/apify/apify-docs/tree/master/examples/ts-parallel-runs-scraping-scraper).
```
Comment (TC-MO) [inline sources/academy/tutorials/node_js/multiple-runs-scrape.md:198]:
```suggestion
You can check [full code example](https://github.com/apify/apify-docs/tree/master/examples/ts-parallel-runs-scraping-scraper).
```

**Review by metalwarrior665 — CHANGES_REQUESTED** [review]
I could imagine a solution without a master actor where the first run of the scraper would do the mastering as well and call itself with `childRun: true` and pass its queue/dataset. I think we used that approach sometime. But I haven't thought enough if it is really better UX-wise than a separate master.

**Comment by drobnikj** [comment]
> imagine a solution without a master actor

Agree, but this one serves better as an example. It is more readable.

**Comment by drobnikj** [comment]
> let's think if we could possibly later manage this on the SDK/platform side to make every actor using queue/dataset infinitely horizontally scalable

It can be some product feature. We can bundle it into SDK in some generic way and make some tweeks in the console UI to support this. 
cc product @zenisjan - the feature can be named "Better support for horizontal scaling of Actors"

File: .eslintrc.json [hunk 7]
```diff
@@ -66,5 +66,6 @@
                 "@typescript-eslint/promise-function-async": "off"
             }
         }
-    ]
+    ],
+    "ignorePatterns": ["examples/"]
```
[the remark this answers is not in the document]
Comment (drobnikj) [reply .eslintrc.json:70]: The repository had no monorepo setup, so it created some issues. 
The actors itself has lint npm scripts, so the users can easily copy&paste source code and start using it even with linters.

File: examples/ts-parallel-runs-scraping-scraper/src/main.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+            selector: 'a[href]',
+        });
+        log.info(`Enqueued ${newPages.processedRequests.length} new pages.`);
+        // If the product page is loaded, save the title and URL to the Dataset.
+        if (request?.loadedUrl?.includes('/products/')) {
+            const title = $('title').text();
+            await dataset.pushData({ url: request.loadedUrl, title });
+        }
+    },
+});
```
[the remark this answers is not in the document]
Comment (drobnikj) [reply examples/ts-parallel-runs-scraping-scraper/src/main.ts]: I would keep this logic on crawlee. I think it is fine that the scraper finishes once the queue is empty for a while. The locked requests will be handled by the run, which locks them.

**Review by metalwarrior665 — APPROVED** [review]

File: sources/platform/storage/request_queue.md [hunk 9]
```diff
@@ -460,7 +460,7 @@ await requestQueueClientOne.delete();
 await Actor.exit();
 ```
 
-A detailed tutorial on how to crawl one request queue from multiple Actor runs can be found in [Advanced web scraping academy](https://docs.apify.com/academy/advanced-web-scraping/multiple-runs-scrape).
+A detailed tutorial on how to crawl one request queue from multiple Actor runs can be found in [Academy tutorials](https://docs.apify.com/academy/node-js/multiple-runs-scrape).
```
Comment (mtrunkat) [inline sources/platform/storage/request_queue.md:463]:
```suggestion
A detailed tutorial on how to process one request queue with multiple concurrent Actor runs can be found in [Academy tutorials](https://docs.apify.com/academy/node-js/multiple-runs-scrape).
```

IMHO you are not scraping the queue, you are scraping the website and have runs processing the queue.

**Review by mtrunkat — COMMENTED** [review]
just one suggestion on naming in the main article, perhaps there are more places to alligned

File: sources/academy/tutorials/node_js/multiple-runs-scrape.md [hunk 10]
```diff
@@ -0,0 +1,236 @@
+---
+title: Scrape website in parallel with multiple Actor runs
+description: Learn how to run multiple instances of an Actor to scrape a website faster. This tutorial will guide you through the process of setting up your scraper.
+sidebar_position: 15.10
+slug: /node-js/multiple-runs-scrape
+---
+
+# Scraping website from multiple Actor runs
```
Comment (TC-MO) [inline sources/academy/tutorials/node_js/multiple-runs-scrape.md:8]: Is this different from `title` on purpose?

File: sources/academy/tutorials/node_js/multiple-runs-scrape.md [hunk 11]
```diff
@@ -0,0 +1,236 @@
...
+const requestQueue = await Actor.openRequestQueue(requestQueueId);
+const dataset = await Actor.openDataset(datasetId);
+```
+
+Once you initialized the request queue and dataset, you can start scraping the website.
+In this example, we will use the CheerioCrawler to scrape [the example of ecommerce website](https://warehouse-theme-metal.myshopify.com/).
+You can create your scraper from the [Crawlee + Cheerio TypeScript Actor template](https://apify.com/templates/ts-crawlee-cheerio).
+
+<Tabs groupId="main">
+<TabItem value="main.ts" label="main.ts">
```
Comment (TC-MO) [inline sources/academy/tutorials/node_js/multiple-runs-scrape.md:198]: can we switch the order of code tabs here for consistency?

**Review by TC-MO — APPROVED** [review]

File: sources/academy/tutorials/node_js/multiple-runs-scrape.md [hunk 12]
```diff
@@ -0,0 +1,236 @@
...
+import ScraperActorMainTs from '!!raw-loader!../../../../examples/ts-parallel-scraping/scraper/src/main.ts';
+import ScraperActorInputSchemaJson from '!!raw-loader!../../../../examples/ts-parallel-scraping/scraper/.actor/input_schema.json';
+
+Imagine a large website that you need to scrape. You have a scraper that works well, but scraping the whole website is slow.
+You can speed up the scraping process by running multiple instances of the scraper in parallel.
+This tutorial will guide you through setting up your scraper to run multiple instances in parallel.
+
+:::note In a rush?
+
+You can check [full code example](https://github.com/apify/apify-docs/tree/master/examples/ts-parallel-scraping/orchestrator) right away.
```
Comment (fnesveda) [inline sources/academy/tutorials/node_js/multiple-runs-scrape.md:28]: Here a link to the whole folder with both Actors is better.
```suggestion
You can check [full code example](https://github.com/apify/apify-docs/tree/master/examples/ts-parallel-scraping) right away.
```

File: sources/academy/tutorials/node_js/multiple-runs-scrape.md [hunk 13]
```diff
@@ -0,0 +1,236 @@
...
+
+The Orchestrator Actor orchestrates the parallel execution of scraper Actor runs.
+It runs multiple instances of the scraper Actor and passes the request queue and dataset to them.
+For the Actor's base structure, we use Apify CLI and create a new Actor with the following command and use the [Empty TypeScript Actor template](https://apify.com/templates/ts-empty).
+
+```bash
+apify create orchestrator-actor
+````
+
+If you don't have Apify CLI installed, check out our installation [instructions](https://docs.apify.com/cli).
```
Comment (fnesveda) [inline sources/academy/tutorials/node_js/multiple-runs-scrape.md:47]:
```suggestion
If you don't have Apify CLI installed, check out our installation [instructions](https://docs.apify.com/cli/docs/installation).
```

**Review by fnesveda — APPROVED** [review]
Looks good, just two link fixes, plus I agree with the others' suggestions.

**Comment by TC-MO** [comment]
Is anything halting release of this tutorial?

**Pull request merged**